doc/whatsnew/2.6.rst

```diff
@@ -9,6 +9,7 @@
 Summary -- Release highlights
 =============================
 
+* Python 3.6+ is now required.
```

Comment:
Should this be move to `whatsnew/2.7.rst`?

Reply:
You're right we definitely need to check all the whats new here and also in another MR before releasing.

Reply:
I'm going to do this in another MR, this need discussion that would block this MR.